packages/vite/src/node/optimizer/registerMissing.ts

```diff
@@ -41,40 +52,76 @@ export function createMissingImporterRegisterFn(
   }
 
   let enqueuedRerun: (() => void) | undefined
-  let currentlyProcessing = true
-  initialProcessingPromise.then(() => {
-    currentlyProcessing = false
-    enqueuedRerun?.()
-  })
+  let currentlyProcessing = false
 
-  async function rerun(ssr: boolean | undefined) {
-    // debounce time to wait for new missing deps finished, issue a new
-    // optimization of deps (both old and newly found) once the previous
-    // optimizeDeps processing is finished
+  // If there wasn't a cache or it is outdated, perform a fast scan with esbuild
+  // to quickly find project dependencies and do a first optimize run
+  if (!cachedMetadata) {
+    currentlyProcessing = true
+
+    const scanPhaseProcessing = newDepOptimizationProcessing()
+    optimizedDeps.scanProcessing = scanPhaseProcessing.promise
+
+    const warmUp = async function () {
```

Comment:
Any reason why using `async function () {` over `async () => {`?
If it's for debugging, we may want to name it, then we would just have `async function warmUp() {` without `const warmUp =` in front of it

Reply:
Replacing for `async () => {`, good catch